tests/scripts/test_grow_nexus.py

```diff
@@ -0,0 +1,71 @@
+import os
+import tempfile
+
+import h5py
+import numpy as np
+import pytest
+
+from ess.reduce.scripts.grow_nexus import grow_nexus_file
+
+
+@pytest.fixture
+def nexus_file():
+    with tempfile.TemporaryDirectory() as tmp:
+        path = os.path.join(tmp, 'test.nxs')
+        with h5py.File(path, 'a') as hf:
+            entry = hf.create_group('entry')
+            entry.attrs['NX_class'] = 'NXentry'
+
+            instrument = entry.create_group('instrument')
+            instrument.attrs['NX_class'] = 'NXinstrument'
+
+            detector = instrument.create_group('detector')
+            detector.attrs['NX_class'] = 'NXdetector'
+
+            event_data = detector.create_group('event_data')
+            event_data.attrs['NX_class'] = 'NXevent_data'
+
+            event_data.create_dataset(
+                'event_index',
+                data=np.array([2, 4, 6]),
+                maxshape=(None,),
+                chunks=True,
+            )
+            event_data.create_dataset(
+                'event_time_zero',
+                data=np.array([0, 1, 2]),
+                maxshape=(None,),
+                chunks=True,
+            )
+            event_data.create_dataset(
+                'event_id',
+                data=np.array([0, 1, 2, 0, 1, 2]),
+                maxshape=(None,),
+                chunks=True,
+            )
+            event_data.create_dataset(
+                'event_time_offset',
+                data=np.array([1, 2, 1, 2, 1, 2]),
+                maxshape=(None,),
+                chunks=True,
+            )
+
+            yield hf
+
+
+def test_grow_nexus_unchanged(nexus_file):
+    grow_nexus_file(nexus_file, 1.0, 1.0)
+    assert nexus_file['entry/instrument/detector/event_data/event_index'].size == 3
+    assert nexus_file['entry/instrument/detector/event_data/event_id'].size == 6
+
+
+def test_grow_nexus_double(nexus_file):
+    grow_nexus_file(nexus_file, 2.0, 2.0)
+    assert nexus_file['entry/instrument/detector/event_data/event_index'].size == 3
```

Comment:
Can you check the actual values of the `event_index` instead of just the length?